docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,854 @@
+```
+
+- Get execution context that has recent block IDs, recent transaction IDs, next prices etc.
+
+```go
+context, err := vm.ExecutionContext(nextTime, parent)
+```
+
+- Gets new block, based on the above information
+
+```go
+b := NewBlock(vm, parent, nextTime, context)
+```
+
+- Prune the mempool that belongs to recent blocks
+
+```go
+mempool.Prune(context.RecentBlockIDs)
+```
+
+- Loads the transaction from mempool to aspiring block. It iterates through all the mempool transactions but ignores transactions whose gas units are [exceeding](https://github.com/ava-labs/blobvm/blob/master/chain/builder.go#L65) the remaining block limit, or whose price is [less](https://github.com/ava-labs/blobvm/blob/master/chain/builder.go#L59) that this block's price.
+
+```go
+for mempool.Len() > 0 {
+    next, price := mempool.PopMax()
+    if price < b.Price {
+        mempool.Add(next)
+        log.Debug("skipping tx: too low price", "block price", b.Price, "tx price", price)
+        break
+    }
+    nextLoad := next.LoadUnits(g)
+    if units+nextLoad > g.MaxBlockSize {
+        unusableTxs = append(unusableTxs, next)
+        log.Debug("skipping tx: too large", "block size", units, "tx load", nextLoad)
+        continue // could be txs that fit that are smaller
+    }
+    // Verify that changes pass
+    tvdb := versiondb.New(vdb)
+    if err := next.Execute(g, tvdb, b, context); err != nil {
+        log.Debug("skipping tx: failed verification", "err", err)
+        continue
+    }
+    if err := tvdb.Commit(); err != nil {
+        return nil, err
+    }
+    b.Txs = append(b.Txs, next)
+    units += nextLoad
+}
+```
+
+- Once transactions are added to the block, it will initialize the block to have the marshaled representation and block hash. It will also verify the block.
+
+```go
+if err := b.init(); err != nil {
+    return nil, err
+}
+_, _, err = b.verify()
+```
+
+Finally, it will return the block to the caller, which is the VM and hence the consensus engine.
+
+### Storage
+
+[Storage](https://github.com/ava-labs/blobvm/blob/master/chain/storage.go) handles all the database operations like storing transactions, blocks, account balance etc. Everything is stored as key-value pair for all types (block, transaction, balance etc.) of data.
+
+We prefix different types of key with a unique byte. For eg. block identifier is prefixed with `0x0` and transaction ID with `0x1`. Similarly we have [prefix](https://github.com/ava-labs/blobvm/blob/master/chain/storage.go#L56) for other types as well. The prefix and original ID is separated by a `ByteDelimiter`. Prefixing is necessary for identifying the type of raw byte a paticular key is pointing to.
+
+```go
+const (
+	blockPrefix   = 0x0
+	txPrefix      = 0x1
+	txValuePrefix = 0x2
+	keyPrefix     = 0x3
+	balancePrefix = 0x4
+
+	linkedTxLRUSize = 512
+
+	ByteDelimiter byte = '/'
+)
+```
+
+We have 5 types of key-value pairs in total -
+* **Block** - For storing block data.
```

Comment:
MD004/ul-style: Unordered list style [Expected: dash; Actual: asterisk]markdownlint[MD004](https://github.com/DavidAnson/markdownlint/blob/v0.25.1/doc/Rules.md#md004)

MD032/blanks-around-lists: Lists should be surrounded by blank linesmarkdownlint[MD032](https://github.com/DavidAnson/markdownlint/blob/v0.25.1/doc/Rules.md#md032)